src/game/game_text.hpp

```diff
@@ -12,7 +12,20 @@
 
 #include "../core/smallvec_type.hpp"
 
+struct StringParam {
+	enum ParamType {
+		RAW_STRING,
+		STRING,
+		OTHER
+	} type;
```

Comment:
I'm not really fond of this construct, as it's reminds me of the old `typedef enum { ... } name;` paradigm in C.
I'd rather see the enum definition separate from the field definition. I'd personally even put an empty line between types, fields and functions.